### PR TITLE
Fixed empty product slug bug

### PR DESCRIPTION
**Pull request by Zales0123**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #3011 
| License       | MIT
| Doc PR        |

After passing empty string as product slug, NonUniqueResultException was thrown, due to "findOneBy" method. Now it simply does not generate route, as for invalid slug.

However, it should be considered to generate some "invalidate product" page, rather than "No route found" exception :)

### CODE REVIEW

**Comment by michalmarcinkowski** [comment]
@Zales0123 the proper fix will be to add one if statement before calling repository method:
```php
...
$value = trim($value, '/');
if (empty($value)) {
    continue;
}
$entity = $repository->findOneBy(array($this->routeConfigs[$className]['field'] => $value));
...
```

and please add Behat scenario for that :wink:

File: src/Sylius/Bundle/CoreBundle/Routing/RouteProvider.php [hunk 1]
```diff
@@ -134,13 +134,13 @@ public function getRouteCollectionForRequest(Request $request)
             ) {
                 $value = substr($path, strlen($this->routeConfigs[$className]['prefix']));
                 $value = trim($value, '/');
-                $entity = $repository->findOneBy(array($this->routeConfigs[$className]['field'] => $value));
+                $entity = $repository->findBy(array($this->routeConfigs[$className]['field'] => $value));
 
-                if (!$entity) {
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/CoreBundle/Routing/RouteProvider.php]: You can also change it to:
```php
if (null === $entity) {
```

**Comment by Zales0123** [comment]
Sure, you're right! I have sometimes problem with complicating things to much ;)

**Comment by pjedrzejewski** [comment]
@Zales0123 This scenario should be about 404, we do not want to show exception to the user...

**Comment by pjedrzejewski** [comment]
Thank you Mateusz!

**Pull request merged**